A29-xds-tls-security.md

```diff
@@ -0,0 +1,446 @@
+using an xDS managed channel or server. Not using Xds Channel or Server Credentials results
+in not using xDS-provided TLS configuration.
+
+On the client side, an XdsChannelCredentials is used and on the server side an
+XdsServerCredentials is used as described in the example snippets below. In these examples,
+a plaintext or insecure credentials is used as the fallback credentials.
+
+
+#### C++ 
+
+Use of XdsChannelCredentials:
+
+```C++
+    std::shared_ptr<grpc::ChannelCredentials> credentials =
+        grpc::experimental::XdsCredentials(grpc::InsecureChannelCredentials());
+    std::shared_ptr<grpc::Channel> channel = grpc::CreateChannel(target, credentials);
+```
+
+Use of XdsServerCredentials:
+
+```C++
+   grpc::experimental::XdsServerBuilder builder;
+   builder.RegisterService(&service);
+   builder.AddListeningPort(listening_address,
+                                 grpc::experimental::XdsServerCredentials(
+                                     grpc::InsecureServerCredentials()));
+   builder.BuildAndStart()->Wait();
+```
+
+#### Java
+
+Use of XdsChannelCredentials:
+
+```Java
+    ChannelCredentials credentials = XdsChannelCredentials.create(InsecureChannelCredentials.create());
+    ManagedChannel channel = Grpc.newChannelBuilder(target, credentials).build();
+```
+
+Use of XdsServerCredentials:
+
+```Java
+    ServerCredentials credentials = XdsServerCredentials.create(InsecureServerCredentials.create());
+    Server server = XdsServerBuilder.forPort(port, credentials)
+      .addService(new HostnameGreeter(hostname))
+      .build()
+      .start();
+```
+
+#### Go
+
+Use of XdsChannelCredentials:
+
+```Go
+   import (
+           xdscreds "google.golang.org/grpc/credentials/xds"
+   )
+
+   if creds, err := xdscreds.NewClientCredentials(xdscreds.ClientOptions{FallbackCreds: insecure.NewCredentials()}); err != nil {
+           log.Fatalf("Failed to create xDS client credentials: %v", err)
+   }
+   conn, err := grpc.Dial(*target, grpc.WithTransportCredentials(creds))
+```
+
+Use of XdsServerCredentials:
+
+```Go
+   import (
+           xdscreds "google.golang.org/grpc/credentials/xds"
+           "google.golang.org/grpc/xds"
+   )
+
+   if creds, err := xdscreds.NewServerCredentials(xdscreds.ServerOptions{FallbackCreds: insecure.NewCredentials()}); err != nil {
+           log.Fatalf("Failed to create xDS server credentials: %v", err)
+   }
+   server := xds.NewGRPCServer(grpc.Creds(creds))
+```
+
+### xDS Protocol
+
+gRPC uses CDS to configure client side security and LDS for server side security as described below.
+The required fields were added to xDS v3 hence xDS v3 is a pre-requisite for this proposal.
+
+#### CDS for Client Side Security
+
+The CDS policy is the top level LB policy applied to all the connections under that policy.
+[A27:CDS][] describes the flow. The field [`transport_socket`][CL-TS] is used to extract the
+[`UpstreamTlsContext`][UTC] as described [here][CL-TS-comment].
```

Comment:
The comment talks about setting the `name` field to `tls` and the `typed_config` field to an `UpstreamTlsContext` proto. 

gRPC-Go NACKs resources which contain other values in these fields. Do the other two languages do the same? 
And if so, we should highlight that here.
And if not, what are y'all doing?

Reply:
Same applies to the server side as well.

Reply:
Yes, you have the same point made for the comment for line 178. Let's discuss it there and close this thread?